contracts/WildcardSteward_v2.sol

```diff
@@ -450,13 +470,57 @@ contract WildcardSteward_v2 is Initializable {
         emit RemainingDepositUpdate(patron, deposit[patron]);
     }
 
-    function buy(uint256 tokenId, uint256 _newPrice, uint256 _deposit)
+    function changeAuctionParamters(
+        uint256 _auctionStartPrice,
+        uint256 _auctionEndPrice,
+        uint256 _auctionLength
+    ) external onlyAdmin {
+        require(
+            _auctionStartPrice > _auctionEndPrice,
+            "Auction value must decrease over time"
+        );
+        require(_auctionLength >= 86400, "Auction should last at least day");
+
+        auctionStartPrice = _auctionStartPrice;
+        auctionEndPrice = _auctionEndPrice;
+        auctionLength = _auctionLength;
+    }
+
+    // NOTE: Need to distinguish been previously owned and foreclosed vs first sale items.
+    //
+    function _auctionPrice(uint256 tokenId) internal returns (uint256) {
+        // HERE we need to be careful. Check whether timeLastCollected[tokenId] is actually now due to collect Patronage modifer being called
```

Comment:
When a token has been owned and foreclosed, Instead of using auctionStartPrice, use the price at which the token foreclosed as the auction start price. This means the foreclosure function should probably no longer set the token price to 0, and we can just use price[tokenId] for this.